crypto/zk/prover/prover.go

```diff
@@ -21,10 +21,59 @@ var (
 	ErrWitnessCalc     = fmt.Errorf("error during witness calculation")
 	ErrProofGen        = fmt.Errorf("error during zksnark proof generation")
 	ErrParseProofData  = fmt.Errorf("error parsing the proof provided, it must be a valid json, check https://github.com/iden3/go-rapidsnark/blob/73d5784d2aa791dd6646142b0017dbef97240f57/types/proof.go")
-	ErrParsePubSignals = fmt.Errorf("error during zksnark proof generation, it must be a json with array of strings")
+	ErrParsePubSignals = fmt.Errorf("error during zksnark public signals proof generation, it must be a json with array of strings")
+	ErrEncodingProof   = fmt.Errorf("error encoding prove result into a proof struct")
+	ErrDecodingProof   = fmt.Errorf("error decoding prove as []byte")
 	ErrVerifyProof     = fmt.Errorf("error during zksnark verification")
 )
 
+// ProofData struct contains the calculated parameters of a Proof. It allows to
+// encode and decode go-rapidsnark inputs and outputs easily.
+type ProofData struct {
+	A []string   `json:"pi_a"`
```

Comment:
As far as I understand, these are hexadecimal strings. A better option might be to use our `types.HexBytes{}` which is a []byte wrapper that marshals the bytes as hexadecimal (instead of base64) and has a `.String()` method to transform the byte slice to hex string.

Reply:
Are you suggesting that i should convert each item of each ProofData parameter into a `types.HexBytes{}` or just the inputs of `prover.ParseProof()` and results of `prover.Bytes()` functions? 
I ask this because if each parameter must be a `type.HexByte{}`, this can complicate the code quite a bit, since they should be simple or nested lists.

Reply:
Ideally, the type would be `types.HexBytes{}`, but you're right it requires an extra logic which might not be worth it. So, it's up to you to decide.

Reply:
I will check the possibility of change it to `types.HexBytes{}` and i will post a comment with the result.